catalog/dags/elasticsearch_cluster/create_proportional_by_source_staging_index/create_proportional_by_source_staging_index_dag.py

```diff
@@ -0,0 +1,188 @@
+    doc_md=__doc__,
+    params={
+        "media_type": Param(
+            default=AUDIO,
+            enum=MEDIA_TYPES,
+            description="The media type for which to create the index.",
+        ),
+        "percentage_of_prod": Param(
+            default=0.5,
+            type="number",
+            exclusiveMinimum=0,
+            maximum=1,
+            description=(
+                "The proportion of items to take of each provider from"
+                " the total amount existing in the source index."
+            ),
+        ),
+        "source_index": Param(
+            default=None,
+            type=["string", "null"],
+            description=(
+                "Optionally, the existing staging Elasticsearch index"
+                " to use as the basis for the new index. If not provided,"
+                " the index aliased to `<media_type>-filtered` will be used."
+            ),
+        ),
+    },
+    render_template_as_native_obj=True,
+)
+def create_proportional_by_source_staging_index():
+    # Fail early if any conflicting DAGs are running
+    prevent_concurrency = prevent_concurrency_with_dags(
+        external_dag_ids=[
+            STAGING_DB_RESTORE_DAG_ID,
+            RECREATE_STAGING_INDEX_DAG_ID,
+            CREATE_NEW_INDEX_CONFIGS[STAGING].dag_id,
+        ]
+    )
+
+    es_host = es.get_es_host(environment=STAGING)
+
+    source_index_name = create_index.get_source_index(
+        source_index="{{ params.source_index }}",
+        media_type="{{ params.media_type }}",
+    )
+
+    source_config = es.get_index_configuration(
+        source_index=source_index_name, es_host=es_host
+    )
+
+    destination_index_name = create_index.get_destination_index_name(
+        media_type="{{ params.media_type }}",
+        current_datetime_str="{{ ts_nodash }}",
+        percentage_of_prod="{{ params.percentage_of_prod }}",
+    )
+
+    destination_alias = create_index.get_destination_alias(
+        media_type="{{ params.media_type }}"
+    )
+
+    destination_index_config = create_index.get_destination_index_config(
+        source_config=source_config, destination_index_name=destination_index_name
+    )
+
+    new_index = es.create_index(index_config=destination_index_config, es_host=es_host)
+
+    staging_source_counts = create_index.get_staging_source_counts(
+        source_index=source_index_name, es_host=es_host
+    )
+
+    desired_source_counts = create_index.get_proportional_source_count_kwargs.override(
+        task_id="get_desired_source_counts"
+    )(
+        staging_source_counts=staging_source_counts,
+        percentage_of_prod="{{ params.percentage_of_prod }}",
+    )
+
+    reindex = es.trigger_and_wait_for_reindex.partial(
+        destination_index=destination_index_name,
+        source_index=source_index_name,
+        timeout=timedelta(hours=12),
+        requests_per_second=Variable.get(
+            "ES_INDEX_THROTTLING_RATE", 20_000, deserialize_json=True
+        ),
```

Comment:
Do you think we can change this to a [Jinja templated variable get](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/variables.html) so we're not making DB calls at DAG parse time for it?